alica_engine/src/engine/RunnableObject.cpp

```diff
@@ -52,43 +82,39 @@ void RunnableObject::start(RunningPlan* rp)
     }
     ++_signalState;
     _signalContext.store(rp);
-    if (_requiresParameters) {
-        assert(rp->getParent());
-        assert(rp->getParent()->getBasicPlan());
-        const auto& wrappers = rp->getParent()->getActiveState()->getConfAbstractPlanWrappers();
-        auto it =
-                std::find_if(wrappers.begin(), wrappers.end(), [this](const auto& wrapper_ptr) { return wrapper_ptr->getAbstractPlan()->getName() == _name; });
-        assert(it != wrappers.end());
-
-        int64_t wrapperId = (*it)->getId();
-
-        BasicPlan* parentPlan = rp->getParent()->getBasicPlan();
-        auto& planAttachment = parentPlan->getPlanAttachment(wrapperId);
-        auto initCall = [this, &planAttachment, parentPlan = parentPlan]() {
-            if (!_blackboard) {
-                _blackboard = std::make_shared<Blackboard>();
-            }
-            _blackboard->impl().clear();
-            if (!planAttachment->setParameters(*parentPlan->getBlackboard(), *_blackboard)) {
-                std::cerr << "Setting parameters failed, supposedly as the context has already changed.  Plan will not be scheduled" << std::endl;
-                return;
-            }
-            doInit();
-        };
-        _engine->editScheduler().schedule(initCall);
+    if (rp->getParent()) {
+        if (!_inheritBlackboard) {
+            assert(rp->getParent()->getBasicPlan());
+            const auto& wrappers = rp->getParent()->getActiveState()->getConfAbstractPlanWrappers();
+            auto it = std::find_if(
+                    wrappers.begin(), wrappers.end(), [this](const auto& wrapper_ptr) { return wrapper_ptr->getAbstractPlan()->getName() == _name; });
+            assert(it != wrappers.end());
+
+            int64_t wrapperId = (*it)->getId();
+
+            BasicPlan* parentPlan = rp->getParent()->getBasicPlan();
+            const auto keyMapping = parentPlan->getKeyMapping(wrapperId);
+            auto initCall = [this, keyMapping, parentPlan = parentPlan]() {
+                assert(_blackboard);
+                keyMapping.setInput(parentPlan->getBlackboard().get(), _blackboard.get());
```

Comment:
I think we still need to reset child blackboard here.  The child may have protected parameters that have leftover values from previous runs.

I guess `clear()` needs to be replaced with `reset()` uch that it does not remove keys, but just resets everything to their default values